fe1-web/src/core/objects/Base64UrlData.ts

```diff
@@ -43,4 +45,37 @@ export class Base64UrlData extends String {
     }
     return paddedStr;
   }
+
+  /**
+   * Returns *some* string representation of this object.
+   * If you need access to the unterlying data type use .valueOf() and
+   * if you want to serialize an instance use .toState() instead
+   */
+  public toString(): string {
+    return super.toString();
+  }
+
+  /**
+   * Returns the primitive value used for representing the Base64UrlData,
+   * a string
+   * If you want to serialize an instance use .toState() instead
+   */
+  public valueOf(): string {
+    return super.valueOf();
+  }
```

Comment:
this is already defined in `String`, that we inherit. What's the advantage of re-implementing it?

Reply:
Only the minor advantage of having a comment indicating the usage and it would allow us to change it to "Base64Url(...)" which might underline the different use cases for the functions?

Reply:
While for Hash and Base64Url it does not make a difference I think it reduces the mental load when reading and writing code when we try to consistently use the appropriate variants as for other data types there are real differences between the functions.

Reply:
Fair enough, I can see the argument.
I'm however slightly concerned by the ease with which `toString` is called implicitly by javascript and might fail in subtle ways. If we're comparing a Hash and a Base64Url data , and they contain the same thing, both `toString` should return the same string... otherwise some comparisons might fail.
See https://stackoverflow.com/questions/50036448/when-is-tostring-get-called-implicitly-in-javascript for an example (agreed: strict equalities should be preferred, but this codebase is worked on by many students...).

Reply:
Uhhhhh.. I did not know about `==` calling `toString()`, that is terrible and definitely good to know, thank you 😂 I mean for the specific case of comparisons we fortunately already have  a linter rule forbidding it (https://eslint.org/docs/latest/rules/eqeqeq) but I can definitely see your point! I'll remove the `toString()` methods for the classes inheriting from String 👍

Reply:
Agreed. JavaScript is one backstabbing son of a bad programming language 😅 

If you want some pretty good (and hilarious) examples of JavaScript's beautiful counter-intuitiveness ... https://www.destroyallsoftware.com/talks/wat

Reply:
Haha hilarious indeed, thank you 🤣